src/pybind/mgr/rook/rook_cluster.py

```diff
@@ -203,86 +209,81 @@ def add_filesystem(self, spec):
         # TODO warn if spec.extended has entries we don't kow how
         #      to action.
 
-        rook_fs = {
-            "apiVersion": self.rook_env.api_name,
-            "kind": "CephFilesystem",
-            "metadata": {
-                "name": spec.name,
-                "namespace": self.rook_env.namespace
-            },
-            "spec": {
-                "onlyManageDaemons": True,
-                "metadataServer": {
-                    "activeCount": spec.count,
-                    "activeStandby": True
-
-                }
-            }
-        }
+        rook_fs = cfs.CephFilesystem(
+            apiVersion=self.rook_env.api_name,
+            metadata=dict(
+                name=spec.name,
+                namespace=self.rook_env.namespace,
+            ),
+            spec=cfs.Spec(
+                metadataServer=cfs.MetadataServer(
+                    activeCount=spec.count,
+                    activeStandby=True
+                )
+            )
+        )
 
         with self.ignore_409("CephFilesystem '{0}' already exists".format(spec.name)):
-            self.rook_api_post("cephfilesystems/", body=rook_fs)
+            self.rook_api_post("cephfilesystems/", body=rook_fs.to_json())
 
     def add_nfsgw(self, spec):
         # TODO use spec.placement
         # TODO warn if spec.extended has entries we don't kow how
         #      to action.
 
-        rook_nfsgw = {
-            "apiVersion": self.rook_env.api_name,
-            "kind": "CephNFS",
-            "metadata": {
-                "name": spec.name,
-                "namespace": self.rook_env.namespace
-            },
-            "spec": {
-                "rados": {
-                    "pool": spec.extended["pool"]
-                },
-                "server": {
-                    "active": spec.count,
-                }
-            }
-        }
+        rook_nfsgw = cnfs.CephNFS(
+            apiVersion=self.rook_env.api_name,
+            metadata=dict(
+                name=spec.name,
+                namespace=self.rook_env.namespace,
+            ),
+            spec=cnfs.Spec(
+                rados=cnfs.Rados(
+                    pool=spec.pool
```

Comment:
Shouldn't this still be `spec.extended["pool"]`? Maybe I'm missing it, but I don't see how the spec object passed to `add_nfsgw` changed.

Reply:
this will be changed in https://github.com/ceph/ceph/pull/29063/files#diff-45db35bc78bc3027f9421bd371b19e9bR239 , but that's not yet in master, right.

Reply:
Ah, I see.